### PR TITLE
[Finishes #187355083] Added the postgress and the sequilizer configur…

### PR DESCRIPTION
**Pull request by Christian-Ishimwe**
_### Description_

This pull request aims to use postgres as our SQL database as well as using Sequelizer as our ORM Tool

1. **Dependency Installation:**
 `npm install`

2. **File Structure:**
 Our sequilezer follows this directory:  

   ```
       src/database/
         |- models/   # Directory to store all your database models
         |- migrations/  # Directory to store migration files for database schema changes
         |- seeds/      # Directory to store seed data for testing or initial population
         |- config/ config.ts # File for Sequelize configuration and connection setup
        
     ```

3. **Configuration:**
   - Created a `.env` file in the project root directory to store sensitive database credentials securely.
   - follow .env.example  to create .env file

4. **Sequelize Initialization:**
   - Created `database.ts` to manage Sequelize configuration and connection.

5. **Model Example (Optional):**
   - Provided an example of defining a simple `Admin` model in `models/admins.ts` 

7. **Sequelizer-cli**
refer to package.json for using the Sequelizer-cli commands on bash, ex: f
or migrations.
```shell
npm run migrate
``` 
**Note**

-By using sequelizer-cli, you will get the .js files, hence you will need to change them manually to .ts file.
-In .sequelizerc file, we proveded the paths for sequelizer-cli, while in ` 'config': path.resolve('./src/database/config', 'config.js'),` we referenced in dist directorym which mean you will be required to run this script: /
**Testing**

```bash
npm run build
``` 
Please review and test this initial setup. Further development and testing are required for full integration into the API.

**Additional links**

1. Sequilezer cli Documentation: https://sequelize.org/docs/v6/other-topics/migrations/#configuration
2. SQL Commands:  https://www.w3schools.com/sql/default.asp

### CODE REVIEW

File: README.md
Comment (UmuhireJessie) [inline README.md:1]: Readme files are for providing guideline on the whole project, not specifically to a particular features. Unless it is the main feature. You can provide us details about the chore by updating description section on pull request instead.

File: src/database/database.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Sequelize } from 'sequelize';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const sequelize = new Sequelize(`${process.env.DATABASE_URL}`, {
```
Comment (UmuhireJessie) [inline src/database/database.ts]: Add necessary environment variables in `.env.example` file, e.g., NODE_ENV, DATABASE_URL, TEST_DATABASE_URL, PROD_DATABASE_URL and provide one sentence explaining each

File: src/models/userModels.ts
Comment (UmuhireJessie) [inline src/models/userModels.ts:1]: Please ensure that configure sequelize correctly, we should be having a structure like the following:
- src/database/config
- src/database/migrations
- src/database/models
- src/database/seeders

Also provide a sample code for each files, create a user seeder with an initial admin. [Here](https://sequelize.org/docs/v6/other-topics/migrations/) is the guideline for the above

File: package.json
Comment (UmuhireJessie) [inline package.json:1]: update the script sections, with the scripts to run migrations, undo migration, seeders files undo seeds.
Check [here](https://sequelize.org/docs/v6/other-topics/migrations/)

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Hello @Christian-Ishimwe, please check my comments

File: .env.example
Comment (UmuhireJessie) [inline .env.example:1]: Add NODE_ENV variable

File: .gitignore [hunk 2]
```diff
@@ -25,3 +25,7 @@ dist/
 /node_modules
 .env
 /package-lock.json
+
+
+#database Migrations
+src/database/migrations
```
Comment (UmuhireJessie) [inline .gitignore]: You do not have to ignore migrations folder, this a crucial file while setting up a database

File: README.md [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
  # e-commerce-ones-and-zeroes-bn
 Ones-and-Zeroes e-commerce backend
+POSTGRESS SEQUILIZER
 
+Description
```
Comment (UmuhireJessie) [inline README.md]: Prefer providing us a descriptive Readme, if you want. Or else remove unnecessary details

File: package.json
Comment (UmuhireJessie) [inline package.json:1]: Change scripts sections. [Here](https://sequelize.org/docs/v6/other-topics/migrations/) is a guide

File: src/database/config/config.ts
Comment (UmuhireJessie) [inline src/database/config/config.ts:1]: Remove this folder

File: src/config/config.json
Comment (UmuhireJessie) [inline src/config/config.json:1]: Remove this folder

File: src/database/config/config.ts
Comment (UmuhireJessie) [inline src/database/config/config.ts:1]: Prefer using this instead:
```
const dotenv = require('dotenv')

dotenv.config()
module.exports = {
  development: {
    url: process.env.DATABASE_URL,
    dialect: 'postgres',
    logging: false
  },
  test: {
    url: process.env.TEST_DATABASE_URL,
    dialect: 'postgres',
    logging: false
  },
  production: {
    url: process.env.PROD_DATABASE_URL,
    dialect: 'postgres',
    logging: false
  }
}
```

And add an else if section in `src/database/models/index.ts`
```
let sequelize: any;
if (config.use_env_variable) {
  // @ts-ignore
  sequelize = new Sequelize(process.env[config.use_env_variable], config);
} else if (config.url) {
  // @ts-ignore
  sequelize = new Sequelize(config.url, config)
} else {
  // @ts-ignore
  sequelize = new Sequelize(config.database, config.username, config.password, config);
}
```

This way the code is clean and readable. Then in your `.env.example` you will have to use the following:
```
NODE_ENV=development
DATABASE_URL='Database url for development'
TEST_DATABASE_URL='Database url for test'
PROD_DATABASE_URL='Database url for production'
```

File: src/server.ts [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
+    db.Admins.create(admin)
+  })
+}
 
 dotenv.config();
 
 const app = express();
 const port = process.env.PORT || 7000;
-
+const databaseUrl= process.env.DATABASE_URL
```
Comment (UmuhireJessie) [inline src/server.ts]: How are you having this `DATABASE_URL`, yet the your current configurations didn't have that environment?

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Check my comments

**Comment by Christian-Ishimwe** [comment]
i have corrected the requesed changes, please check out @UmuhireJessie

File: package-lock.json
Comment (UmuhireJessie) [inline package-lock.json:1]: Update scripts section

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@Christian-Ishimwe check my comments. You can not label all the commit with `Finishes`, use that once you have completed everything and that you are ready.

File: .env.example [hunk 5]
```diff
@@ -1 +1,5 @@
-PORT='7000'
```
Comment (UmuhireJessie) [inline .env.example:1]: Add port.

File: package.json [hunk 6]
```diff
@@ -7,15 +7,25 @@
     "build": "tsc",
     "start": "node dist/server.js",
     "dev": "nodemon --exec ts-node src/server.ts",
-    "test": "jest"
+    "test": "jest",
+    "migrate": "sequelize db:migrate",
+    "seed": "sequelize db:seed:all",
+    "undo-migrate": "sequelize db:migrate:undo:all"
```
Comment (UmuhireJessie) [inline package.json]: add a script for undoing seed, and that of doing everything all at once for simplicity.

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
The migrations are not going through!
image

Please do the following:
1. Ensure that you have tested everything before adding reviewers
2. Add a description with how can we test your chore manually, right now everyone is just imaging, what do we expect if the chore is successfully done, etc!
3. Squash the commits!

File: src/database/models/index.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
+const process = require('process');
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.ts')[env];
```
Comment (UmuhireJessie) [inline src/database/models/index.ts]: change these lines with
```
const env = process.env.NODE_ENV || 'development';
const configSuffix = process.env.NODE_ENV === 'development' ? '.ts' : '.js';
const config = require(__dirname + `/../config/config${configSuffix}`)[env];
```

File: src/database/seeders/Admins.seeds.ts
Comment (UmuhireJessie) [inline src/database/seeders/Admins.seeds.ts:1]: This is not how we create seeders. Here is the [guide](https://sequelize.org/docs/v6/other-topics/migrations/#creating-the-first-seed)

File: src/database/models/admins.ts
Comment (UmuhireJessie) [inline src/database/models/admins.ts:1]: The fields you have in you migration file and model file should be the same. If it is firstName, lastName, email and telephone, that is what should be in migration too

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@Christian-Ishimwe please resolve the comments

**Pull request closed without merging**